### ISSUE TITLE
FN — `always-unstable-deps` evaluates deps against an empty heap, so a member dep is always ⊤

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#135]
Found while adding the `@mantine/form` wrapper table (#94 follow-up).

## The defect

`always_unstable_deps::eval_dep_is_unstable` seeded `eval_in_stores` with a
fresh `Heap::new()`. A member read (`obj.f`, `form.onSubmit`) only resolves
through the heap — `eval_field_access` follows the `Loc` into
`HeapValue::Obj`. With an empty seed the read degraded to ⊤, and ⊤ is silent
by design in this rule ("`Top` (precision lost) stays silent to avoid FPs").

So the rule never fired on a member dep, however provably fresh the member was.

```jsx
function C() {
  const obj = { f: () => {} };
  useEffect(() => {}, [obj.f]);   // silent — `obj.f` is a fresh fn every render
}
```

A **false negative**, and the direction the project forbids.

`missing-deps` already seeds `result.heap.clone()` for the same deps arrays
(`src/rules/impls/missing_deps.rs:155`), so the two sibling rules disagreed
about the same expressions.

## Why it was invisible until now

The converged heap only became worth reading recently: #88 gave object literals
a per-member map, and #134 made an allocation site identify one allocation site.
Before those, a member read could answer from the wrong object.

## Fix

Pass `result.heap.clone()`. `is_unstable_reference_only()` is a proof
predicate, so the change can only turn ⊤ into a proven value — it adds
findings, never removes them.

## Acceptance criteria

- [x] `[obj.f]` where `f` is a fresh arrow fires `always-unstable-deps`
- [x] `[obj.n]` where `n` is a number stays silent (value comparison)
- [x] Gate-by-removal: restoring `Heap::new()` reddens exactly the first test
- [ ] Corpus delta measured

**Comment by rboudrouss** [comment]
Fixed in c3a23cd.

## What it turned out to be

Not one rule. `ConvergedEval::eval_in` took the heap as a caller argument, on the theory that an empty and a converged seed were two legitimate choices. Four of its six call sites took the empty one. `redundant-set-state` and `unnecessary-rerender` gate on `is_stable()`, which ⊤ also fails — so they were missing the same family of findings for the same reason.

So the fix is central rather than per-rule: `eval_in` seeds `self.heap.clone()` and the parameter is gone. A site that genuinely evaluates against *empty* stores calls the `eval_in_stores` primitive, and that bundle has no converged half to be inconsistent with. `redundant-set-state`, which threads decomposed stores by hand, now threads one scratch heap with them.

An `Eval` type owns the scratch heap so the seed is neither a per-call decision nor a per-call clone. `missing-deps` already cloned once and reused across a path's prefixes; the first cut of this fix would have regressed that to one clone per prefix. Reuse is sound because an `ExprId` names one allocation site (#134) — re-probing a site rewrites its own entry, and two sites never collide.

## Corpus

**1 325 → 1 325, 0 removed and 0 added**, verified on both sides rather than on the total alone.

The false negative is real — unit test plus gate-by-removal, where restoring `Heap::new()` reddens exactly the member-dep test and nothing else — it simply does not occur in these 14 repos. Recording it as a result: a soundness fix is justified by what it makes impossible, not by what it moves today.

No timing cost: dub measures 75 s / 74 s / 71 s across the three binaries, and the full corpus 827 s → 807 s.

## Acceptance criteria

- [x] `[obj.f]` where `f` is a fresh arrow fires `always-unstable-deps`
- [x] `[obj.n]` where `n` is a number stays silent
- [x] Gate-by-removal reddens exactly the first test
- [x] Corpus delta measured (+0, both sides)
